### PR TITLE
Throw ValueError if shapes don't match in keras.layers.Softmax

### PR DESCRIPTION
**Pull request by szutenberg**
Due to https://github.com/tensorflow/tensorflow/issues/50467
(users may be not aware of how does keras.layers.Masking work)

@gbaned asked me to push this commit here (previous PR: https://github.com/tensorflow/tensorflow/pull/50586 )

### CODE REVIEW

File: keras/layers/advanced_activations.py [hunk 1]
```diff
@@ -338,7 +338,11 @@ def call(self, inputs, mask=None):
 
       # Since we are adding it to the raw scores before the softmax, this is
       # effectively the same as removing these entirely.
-      inputs += adder
+      try:
```
Comment (fchollet) [inline keras/layers/advanced_activations.py:341]: Rather than try/except, do an explicit shape check. Note that if the shape is dynamic, no error can be raised until actual runtime (e.g. the try/except would not work).

When raising an error message:

- Print what the input shapes were (give context about what happened)
- Print what they should have been (state how to fix the issue)
- Make sure to add a unit test that checks the message is raised as expected (`AssertRaises` with a regex)

**Review by fchollet — CHANGES_REQUESTED** [review]

**Comment by gbaned** [comment]
@szutenberg Can you please check @fchollet's comments and keep us posted ? Thanks!

**Comment by szutenberg** [comment]
Hi @gbaned ,

@fchollet 's comments are valid but due to the summer season, they have to wait a bit more ;) I plan to fix it in August.

**Comment by qlzh727** [comment]
Closing this PR since it is staled for a while. @szutenberg feel free to reopen this one when you have time to address the comment from Francois.
